pkg/instaclustr/client.go

```diff
@@ -2291,3 +2291,57 @@ func (c *Client) DeleteAWSEndpointServicePrincipal(principalID string) error {
 
 	return nil
 }
+
+func (c *Client) CreateOpenSearchEgressRules(rule *clusterresourcesv1beta1.OpenSearchEgressRules) error {
+	url := c.serverHostname + OpenSearchEgressRulesEndpoint
+
+	b, err := json.Marshal(rule.Spec)
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.DoRequest(url, http.MethodPost, b)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+	b, err = io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusAccepted {
+		return fmt.Errorf("status code: %d, message: %s", resp.StatusCode, b)
+	}
+	err = json.Unmarshal(b, &rule.Status)
+	if err != nil {
+		return err
+	}
```

Comment:
It would be clearer if you just returned the bytes and only then converted them to `rule.Status`. This shows that the status is indeed changing and you are performing the `patch` operation.

Reply:
After discussion with the team, we agreed that we should convert data in client